things/policies/api/http/transport.go

```diff
@@ -94,8 +94,10 @@ func decodeConnectThing(_ context.Context, r *http.Request) (interface{}, error)
 		GroupID:  bone.GetValue(r, "chanId"),
 		ClientID: bone.GetValue(r, "thingId"),
 	}
-	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
-		return nil, errors.Wrap(errors.ErrMalformedEntity, err)
+	if r.Body != http.NoBody {
```

Comment:
Is there any particular reason to check the body? Giving options (to override) will only result in confusing API.